daffodil-core/src/main/scala/org/apache/daffodil/core/dpath/Expression.scala

```diff
@@ -2295,6 +2295,18 @@ abstract class FunctionCallBase(
     res
   }
 
+  final def checkArgArray(): Unit = {
+    lazy val isArray = expressions.head match {
+      case rpe: RelativePathExpression => rpe.steps.last.isArray
+      case rpe: RootPathExpression => rpe.steps.last.isArray
```

Comment:
Can these two be combined into this?
```scala
case pe: PathExpression => pe.steps.last.isArray
```
I think these are the only classes that extend that base class?

Also, is `isArray` sufficient? I know optional elements (i.e. minOccurs=0 maxOccurs=1) are represented in the infoset as an array, but sometimes aren't considered as actual arrays in other parts of the code. I'm not sure if the StepExpression isArray handles that? I think we want `fn:count` to still work on optional elements?

Reply:
The RelativePathExpression and RootPathExpression cases are now combined as PathExpression.